configmaptocrs/main.go

```diff
@@ -42,57 +43,59 @@ func main() {
 	log.Printf("MetalLB generator starting. commit: %s branch: %s goversion: %s",
 		version.CommitHash(), version.Branch(), version.GoString())
 
-	f, err := os.Create(filepath.Join(inputDirPath, outPutFileName))
+	res, err := generate(*source)
+	if err != nil {
+		log.Fatalf("failed to generate resources: %s", err)
+	}
+	log.Println("Generator finished successfully!")
+
+	f, err := os.OpenFile(*outPutFilePath, os.O_CREATE|os.O_WRONLY, 0666)
 	if err != nil {
 		log.Fatalf("failed to create output file: %s", err)
 	}
 	defer f.Close()
 
-	err = generate(f, *source)
+	_, err = f.Write(res)
```

Comment:
Again, if we stick to always writing to a file, let's revert the change and leave the writer there.

Reply:
done